code/game/objects/systems/cell_slot.dm

```diff
@@ -101,3 +134,83 @@
 	obj_cell_slot.remove_yank_time = 0
 	obj_cell_slot.legacy_use_device_cells = TRUE
 	return obj_cell_slot
+
+//? Wrappers for cell.dm functions
+//+ These exist to sanely perform cell functions through the obj_cell_slot system. These break safely if no cell exists.
+
+/**
+ * cell function wrapper - returns the rating of the cell inside or null
+ */
+/datum/object_system/cell_slot/proc/get_rating()
+	return cell?.get_rating()
+
+/**
+ * cell function wrapper - returns the cell src or null
+ */
+/datum/object_system/cell_slot/proc/get_cell(inducer)
+	return cell?.get_cell(inducer)
+
+/**
+ * cell function passthrough - consumes energy using *universal units*
+ * returns amount used or null
+ */
+/datum/object_system/cell_slot/proc/drain_energy(datum/actor, amount, flags)
+	return cell?.drain_energy(actor, amount, flags)
+
+/**
+ * cell function wrapper - returns % charge of the cell or null
+ */
+/datum/object_system/cell_slot/proc/percent()
+	return cell?.percent()
+
+/**
+ * cell function wrapper - checks if cell is fully charged
+ * returns true or false, or null if no cell
+ */
+/datum/object_system/cell_slot/proc/fully_charged()
+	return cell?.fully_charged()
+
+/**
+ * cell function wrapper - returns true if cell can provide specified amount
+ * returns null if no cell
+ */
+/datum/object_system/cell_slot/proc/check_charge(var/amount)
+	return cell?.check_charge(amount)
+
+/**
+ * cell function wrapper - returns how much charge is missing from the cell or null
+ */
+/datum/object_system/cell_slot/proc/amount_missing()
+	return cell?.amount_missing()
+
+/**
+ * cell function wrapper - attempts to use power from cell, returns the amount actually used or null
+ */
+/datum/object_system/cell_slot/proc/use(var/amount)
+	return cell?.use(amount)
+
+/**
+ * cell function wrapper - checks if the specified amount can be provided. If it can, it removes the amount from the cell and returns TRUE otherwise does nothing and returns FALSE
+ * returns null if no cell
+ */
+/datum/object_system/cell_slot/proc/checked_use(var/amount)
+	return cell?.checked_use(amount)
```

Comment:
If you want to either return true or false and no null
```suggestion
	return cell?.checked_use(amount) ? TRUE : FALSE
```

Reply:
this one is optional and just FYI, the other two I think should be implemented

Reply:
there's two ways to go here.

1. do TRUE / FALSE if it's success / failure to match cell.checked_use
2. add null to allow the caller to check if there's a cell with this proc too using something like `switch(cell_slot.checked_use())`

i'm kind of on the camp of 1 tbh; if something cares about if the cell is there they should be checking `isnull(cell_slot.cell) anyways`

Reply:
same goes for use() and others; if you have a use case for checking for null this way do it but otherwise just || 0 in use() and similar to coalesce to 0 if it's null.

Reply:
Adjusted all the returns for wrappers to return 0 or FALSE depending on context of the function. Moved away from any NULL returns.